remoting/src/main/java/org/apache/rocketmq/remoting/protocol/FastCodesHeader.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.rocketmq.remoting.protocol;
+
+import java.util.HashMap;
+
+import org.apache.rocketmq.remoting.exception.RemotingCommandException;
+
+import io.netty.buffer.ByteBuf;
+
+public interface FastCodesHeader {
+
+    default String getAndCheckNotNull(HashMap<String, String> fields, String field) {
+        String value = fields.get(field);
+        if (value == null) {
+            String headerClass = this.getClass().getSimpleName();
```

Comment:
maybe cached headerClasss is beffer

Reply:
Here is an exception branch, there is no need to cache the headerCalss string.